test/Algebra/Graph/Test/NonEmpty/AdjacencyIntMap.hs

```diff
@@ -0,0 +1,626 @@
+{-# LANGUAGE OverloadedLists, ViewPatterns #-}
+-----------------------------------------------------------------------------
+-- |
+-- Module     : Algebra.Graph.Test.NonEmpty.AdjacencyIntMap
+-- Copyright  : (c) Andrey Mokhov 2016-2019
+-- License    : MIT (see the file LICENSE)
+-- Maintainer : andrey.mokhov@gmail.com
+-- Stability  : experimental
+--
+-- Testsuite for "Algebra.Graph.NonEmpty.AdjacencyIntMap".
+-----------------------------------------------------------------------------
+module Algebra.Graph.Test.NonEmpty.AdjacencyIntMap (
+    -- * Testsuite
+    testNonEmptyAdjacencyIntMap
+    ) where
+
+import Control.Monad
+import Data.Semigroup ((<>))
+import Data.Tree
+import Data.Tuple
+
+import Algebra.Graph.NonEmpty.AdjacencyIntMap
+import Algebra.Graph.Test hiding (axioms, theorems)
+import Algebra.Graph.ToGraph (reachable)
+
+import qualified Algebra.Graph.AdjacencyIntMap          as AIM
+import qualified Algebra.Graph.NonEmpty.AdjacencyIntMap as NonEmpty
+import qualified Data.List.NonEmpty                     as NonEmpty
+import qualified Data.Set                               as Set
+import qualified Data.IntSet                            as IntSet
+
+sizeLimit :: Testable prop => prop -> Property
+sizeLimit = mapSize (min 10)
+
+type G = NonEmpty.AdjacencyIntMap
+
+axioms :: G -> G -> G -> Property
+axioms x y z = conjoin
```

Comment:
Let's factor out these axioms and theorems for non-empty graphs to `Algebra.Graph.Test` too, to avoid code duplication.

Reply:
Does that require `Class Graph` instances for the `NonEmpty` types?

Reply:
To fit with the `GraphTestsuite g` synonym

Reply:
Hmm, I think you are right, it's not that easy. OK, let's keep this as is, but leave a TODO to fix it at some point.